src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstanceEventsController.cs

```diff
@@ -0,0 +1,177 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading.Tasks;
+using Altinn.Platform.Storage.Models;
+using Altinn.Platform.Storage.Repository;
+using Microsoft.AspNetCore.Mvc;
+using Serilog;
+using Serilog.Core;
+
+namespace Altinn.Platform.Storage.Controllers
+{
+    /// <summary>
+    /// API for managing the instance event element
+    /// </summary>
+    [Route("storage/api/v1/[controller]")]
+    public class InstanceEventsController : Controller
+    {
+        private readonly IInstanceEventRepository _repository;
+        private Logger logger = new LoggerConfiguration()
+            .WriteTo.Console()
+            .CreateLogger();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InstanceEventsController"/> class
+        /// </summary>
+        /// <param name="instanceEventRepository">the instance repository handler</param>
+        public InstanceEventsController(IInstanceEventRepository instanceEventRepository)
+        {
+            _repository = instanceEventRepository;
+        }
+
+        /// <summary>
+        /// Inserts new instance event into the instanceEvent collection.
+        /// </summary>
+        /// <param name="instanceEvent">Id of instance to retrieve events for. </param>
+        /// <returns>The stored instance event.</returns>
+        /// POST storage/api/v1/instanceevents/
+        [HttpPost]
+        public async Task<ActionResult> Post([FromBody] InstanceEvent instanceEvent)
+        {
+            if (instanceEvent == null)
+            {
+                return BadRequest("Missing parameter values: instance event must exist and instanceId must be set");
```

Comment:
Missing check for instanceId.  Should database create Id or should client generate guid for Id?

Reply:
Added the check. 
@dskogan  I don't know what would be the best option regarding the Id. The way I see it, we don't use it for anything in particular (except deletion, but thats a whole different story), and if the database can autogenerate it that means less operations in the method and fewer lines of code to maintain.

Reply:
Ids will be generated by cosmos, closing this comment now.